src/packagedcode/pypi.py

```diff
@@ -270,11 +270,13 @@ class PythonSdistPkgInfoFile(BaseExtractedPythonLayout):
 
     @classmethod
     def parse(cls, location):
-        yield parse_metadata(
-            location=location,
-            datasource_id=cls.datasource_id,
-            package_type=cls.default_package_type,
-        )
+        if not PythonEggPkgInfoFile.is_datafile(location) and not\
```

Comment:
Please avoid back-slashes for line continuations. Prefer instead using `()` parens.
```suggestion
        if (not PythonEggPkgInfoFile.is_datafile(location) and not
```

Reply:
Also I think it would be best overriding is_datafile() for this

Reply:
Made the requested changes.